ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/TransactionImple.java

```diff
@@ -1114,6 +1116,40 @@ public String toString ()
 		}
 	}
 
+	public void endAssociation(Xid _tranID, XAResource _theXAResource, int xaState, int txInfoState) throws XAException {
+		List<XAResource> xars = new ArrayList<XAResource>();
```

Comment:
You could save declaring this memory (and the extra processing needed to add to and to iterate through the list) if you put the code from line 1132 through 1150 into a separate private method.